### PR TITLE
Adapter fixes (event types and source)

### LINKED ISSUE #45: New "mongodb.collection change event" types
Let's change the type to be something like: `google.com.mongodb.collection.v1.inserted` for inserts, `google.com.mongodb.collection.v1.deleted` for deletions, and `google.com.mongodb.collection.v1.updated` for updates.

And let's add them to the registry annotation in the yaml, similar to https://github.com/google/knative-gcp/blob/master/config/core/resources/cloudstoragesource.yaml#L23

_Originally posted by @nachocano in https://github.com/googleinterns/knative-source-mongodb/pull/42_

### LINKED ISSUE #46: Update Receive adapter (small adds/fixes)
**Problem**
Small adds/fixes to #17 

**Fixes to be made**
- change the EventSource env variable to be more descriptive than https://github.com/googleinterns/knative-source-mongodb/pull/42#discussion_r464559163
- #45

### PR DESCRIPTION
**Pull request by selim5**
Fixes #45 
Fixes #46

### CODE REVIEW

File: pkg/reconciler/mongodb/mongodbsource.go [hunk 1]
```diff
@@ -248,7 +248,7 @@ func (r *Reconciler) makeEventSource(ctx context.Context, src *v1alpha1.MongoDbS
 		return "", err
 	}
 
-	return fmt.Sprintf("%s/%s", url.Hostname(), src.Spec.Database), nil
+	return fmt.Sprintf("mongodb://%s/databases/%s/collections/%s", url.Hostname(), src.Spec.Database, src.Spec.Collection), nil
```
Comment (nachocano) [inline pkg/reconciler/mongodb/mongodbsource.go:251]: here you need to do an if checking wether spec.Collection is empty, if so, then don't add /collections/""

File: pkg/adapter/adapter.go [hunk 2]
```diff
@@ -152,20 +153,29 @@ func (a *mongoDbAdapter) makeCloudEvent(data bson.M) (*cloudevents.Event, error)
...
 	// 		Source -> Source environment variable.
-	// 		Type   -> type of change either insert, delete or update.
+	// 		Type   -> type of change either inserted, deleted or updated.
 	//		Data   -> data payload containing either id only for
 	//                deletion or full object for other changes.
 	change, err := utils.DecodeChangeBson(data)
 	if err != nil {
 		return nil, fmt.Errorf("error decoding bson change object: %w", err)
 	}
 	event.SetID(change.ID)
```
Comment (nachocano) [inline pkg/adapter/adapter.go:166]: let's do something like this for this one... https://github.com/google/knative-gcp/blob/11a70b6eaacd5e70b36db7da78f6a19057374ff6/pkg/schemas/v1/auditlogs.go#L46

File: pkg/adapter/adapter.go [hunk 3]
```diff
@@ -152,20 +153,29 @@ func (a *mongoDbAdapter) makeCloudEvent(data bson.M) (*cloudevents.Event, error)
...
 	event := cloudevents.NewEvent(cloudevents.VersionV1)
 
-	// Set cloud event specs and attributes. TODO: issue #43
+	// Set cloud event specs and attributes.
 	// 		ID     -> id of mongo change object
 	// 		Source -> Source environment variable.
-	// 		Type   -> type of change either insert, delete or update.
+	// 		Type   -> type of change either inserted, deleted or updated.
 	//		Data   -> data payload containing either id only for
 	//                deletion or full object for other changes.
```
Comment (nachocano) [inline pkg/adapter/adapter.go:161]: I think I'd rather remove all this comment. Is clearer if you just read the code

File: pkg/adapter/adapter.go [hunk 4]
```diff
@@ -152,20 +153,29 @@ func (a *mongoDbAdapter) makeCloudEvent(data bson.M) (*cloudevents.Event, error)
...
 	change, err := utils.DecodeChangeBson(data)
 	if err != nil {
 		return nil, fmt.Errorf("error decoding bson change object: %w", err)
 	}
 	event.SetID(change.ID)
-	event.SetType(change.OperationType)
 	event.SetSource(a.ceSource)
 	event.SetData(cloudevents.ApplicationJSON, change.Payload)
+	switch change.OperationType {
```
Comment (nachocano) [inline pkg/adapter/adapter.go:169]: you can have a map of mongo types to our types and the code would be cleaner... 
but this is optional, not need to do it if you thing it adds complexity

File: pkg/reconciler/mongodb/mongodbsource.go (hunk 1 above)
Comment (selim5) [reply pkg/reconciler/mongodb/mongodbsource.go:251]: Very true!

File: pkg/adapter/adapter.go (hunk 3 above)
Comment (selim5) [reply pkg/adapter/adapter.go:161]: True!

File: pkg/adapter/adapter.go (hunk 4 above)
Comment (selim5) [reply pkg/adapter/adapter.go:169]: done

**Comment by nachocano** [comment]
Thanks @selim5

**Pull request merged**